### PR TITLE
feat(frontend): process view list

### PR DESCRIPTION
**Pull request by Anh-Duy-Tran**
### Linked issue

Closes ducanhpham0312/zeevision-private#91
Closes ducanhpham0312/zeevision-private#115

### Description

* Process List view ultilizes the table component, each process list item is also extendable to its instances list.

https://github.com/ducanhpham0312/zeevision/assets/113171462/313bb33f-0046-420d-bda7-0f129298cd5a

* Style touch up
* Make element in Single Process Page resizable (drag the side handler)

https://github.com/ducanhpham0312/zeevision/assets/113171462/3abbe23b-cc07-439c-8ec7-3bd51c575d69

### Testing

* Start test bench
* Start zeevision
* Initialize database (with the init sql file in #backend channel)
* Go to localhost:8080

### CODE REVIEW

File: frontend/src/pages/SingleProcessPage.tsx [hunk 1]
```diff
@@ -1,56 +1,38 @@
...
-            "Deployment Time",
-          ]}
-          content={
-            data ? [[processKey, bpmnProcessId, version, deploymentTime]] : []
-          }
-        />
-        <BpmnViewer bpmnString={decodedBpmn} />
+      <div className="mb-[40px] flex h-[30vh]">
+        <div className="w-[30vw]">
+          <Table
```
Comment (ducanhpham0312) [inline frontend/src/pages/SingleProcessPage.tsx:23]: A preliminary question: Can we add the tailwind style directly in the <Table> tag or it must be in a separated div like this?

Comment (Anh-Duy-Tran) [reply frontend/src/pages/SingleProcessPage.tsx:23]: yes we can by exposing another props in the Table component to take in className, I'm modifying this file atm so I'll look into that later, my local changes doesn't include that div wrapper anymore

File: frontend/src/components/Table/HorizontalTable.tsx [hunk 2]
```diff
@@ -62,51 +75,95 @@ export function HorizontalTable({ header, content }: HorizontalTableProps) {
...
-                  {item}
+            <th key={item} className="cursor-pointer py-1 text-left">
+              <Button
+                fullWidth
+                className="text-left"
+                onClick={() => handleSort(item)}
+              >
+                <div className="flex justify-between">
+                  <p>{item}</p>
                   {sortBy === item ? (sortOrder === "asc" ? " ▲" : " ▼") : ""}
```
Comment (ducanhpham0312) [inline frontend/src/components/Table/HorizontalTable.tsx:95]: Please move line 95 inside the `<p>` tag. This way we won't need to update some of the test cases.

File: frontend/src/components/Table/table.test.tsx [hunk 3]
```diff
@@ -87,7 +89,8 @@ describe("HorizontalTable Component", () => {
       />,
     );
     fireEvent.click(getByText("Time created"));
-    expect(getByText("Time created ▲")).toBeInTheDocument();
+    expect(getByText("Time created")).toBeInTheDocument();
+    expect(getByText("▲")).toBeInTheDocument();
```
Comment (ducanhpham0312) [inline frontend/src/components/Table/table.test.tsx:93]: This won't have to be updated after the change above

File: frontend/src/components/Table/table.test.tsx [hunk 4]
```diff
@@ -104,8 +107,9 @@ describe("HorizontalTable Component", () => {
       />,
     );
     fireEvent.click(getByText("Time created"));
-    fireEvent.click(getByText("Time created ▲"));
-    expect(getByText("Time created ▼")).toBeInTheDocument();
+    fireEvent.click(getByText("Time created"));
+    expect(getByText("Time created")).toBeInTheDocument();
+    expect(getByText("▼")).toBeInTheDocument();
```
Comment (ducanhpham0312) [inline frontend/src/components/Table/table.test.tsx:112]: This won't have to be changed as well.

File: frontend/src/pages/SingleProcessPage.tsx [hunk 5]
```diff
@@ -1,73 +1,53 @@
...
-  `;
-  const { data } = useQuery(PROCESS, {
-    pollInterval: queryPollIntervalMs,
-  });
+  const { process } = useQueryProcessData(params.id || "");
+
   const {
     processKey,
-    bpmnProcessId,
+    // bpmnProcessId,
```
Comment (ducanhpham0312) [inline frontend/src/pages/SingleProcessPage.tsx:13]: Why is this being commented out? This should be queried and added to the table

File: frontend/src/pages/SingleProcessPage.tsx [hunk 6]
```diff
@@ -1,73 +1,53 @@
...
-      </div>
+    <div className="flex h-full w-full flex-col gap-3">
+      <ResizableContainer direction="vertical">
+        <div className="flex h-full">
+          <ResizableContainer direction="horizontal">
+            <div className="w-full pr-3">
+              <Table
+                orientation="vertical"
+                header={["Process Key", "Version", "Deployment Time"]}
+                content={process ? [[processKey, version, deploymentTime]] : []}
```
Comment (ducanhpham0312) [inline frontend/src/pages/SingleProcessPage.tsx:29]: please add `bpmnProcessId` back

File: frontend/src/components/Table/HorizontalTable.tsx [hunk 7]
```diff
@@ -4,17 +4,30 @@ import {
...
+  content,
+  alterRowColor,
+  expandElement,
+  optionElement,
+}: HorizontalTableProps) {
   const [page, setPage] = React.useState(0);
   const [rowsPerPage, setRowsPerPage] = React.useState(10);
   const [sortBy, setSortBy] = React.useState("");
-  const [sortOrder, setSortOrder] = React.useState("desc");
+  const [sortOrder, setSortOrder] = React.useState("asc");
```
Comment (ducanhpham0312) [inline frontend/src/components/Table/HorizontalTable.tsx:29]: It is now "desc" in master. Please merge `master` to this branch to get the latest updates

**Review by ducanhpham0312 — COMMENTED** [review]
Good work on the UI. Please check the comments to further improve this. But before that, please merge `master` to this branch to get the latest updates.

File: frontend/src/components/Table/HorizontalTable.tsx (hunk 2 above)
Comment (Anh-Duy-Tran) [reply frontend/src/components/Table/HorizontalTable.tsx:95]: Screenshot 2023-11-25 at 23 32 09

I was going for this design (where the sorting arrow indicator is placed at the opposite side as the header text), could it be styled somehow to still achive the design but have the {item} and the arrow icon in the same element?

File: frontend/src/pages/SingleProcessPage.tsx [hunk 8]
```diff
@@ -25,8 +25,17 @@ export default function SingleProcessPage() {
             <div className="w-full pr-3">
               <Table
                 orientation="vertical"
-                header={["Process Key", "Version", "Deployment Time"]}
-                content={process ? [[processKey, version, deploymentTime]] : []}
+                header={[
+                  "Process Key",
+                  "Process Id",
```
Comment (ducanhpham0312) [inline frontend/src/pages/SingleProcessPage.tsx:30]: "BPMN Process ID"

File: frontend/src/components/Table/HorizontalTable.tsx (hunk 2 above)
Comment (ducanhpham0312) [reply frontend/src/components/Table/HorizontalTable.tsx:95]: Hmm I guess you can keep this change then. I don't see any other way except that it should be in separate elements..

**Review by ducanhpham0312 — APPROVED** [review]
A few new things recently added in HorizontalTable that are not yet tested (which you can see in the Codecov comment), but let's do that later. Great work  👍 

Please update the table name like in the comment below and merge this PR.

**Pull request merged**